src/nat/runtime/session.py

```
@@ -37,67 +41,357 @@
+                                                  ref_count=0,
+                                                  lock=asyncio.Lock())
+                self._per_user_builders[user_id] = builder_info
+                logger.debug(f"Created per-user builder for user={user_id}, entry_function={self._entry_function}")
+                return builder_info.builder, builder_info.workflow
+            except Exception as e:
+                logger.error(f"Error creating per-user builder for user {user_id}: {e}", exc_info=True)
+                try:
+                    await builder.__aexit__(None, None, None)
+                except Exception as cleanup_error:
+                    logger.error(f"Error during builder cleanup after failed creation: {cleanup_error}", exc_info=True)
+                raise e
```

Ruff (0.14.7)

Use `raise` without specifying exception name

Remove exception name

(TRY201)